packages/complex-matcher/src/index.js

```diff
@@ -599,33 +599,45 @@ export const parse = parser.parse.bind(parser)
 
 // -------------------------------------------------------------------
 
-const _getPropertyClauseStrings = ({ child }) => {
+const _getPropertyClauseStringsOrRegex = ({ child }) => {
   if (child instanceof Or) {
     const strings = []
     child.children.forEach(child => {
       if (child instanceof StringNode) {
         strings.push(child.value)
       }
+      if (child instanceof RegExpNode) {
+        const unescapedRegexp = child.re.source.replace(/^(\^)|\\|\$$/g, '')
+        if (child.re.source === `^${escapeRegExp(unescapedRegexp)}$`) {
+          strings.push(unescapedRegexp)
+        }
+      }
     })
     return strings
   }
 
   if (child instanceof StringNode) {
     return [child.value]
   }
+  if (child instanceof RegExpNode) {
+    const unescapedRegexp = child.re.source.replace(/^(\^)|\\|\$$/g, '')
+    if (child.re.source === `^${escapeRegExp(unescapedRegexp)}$`) {
+      return [unescapedRegexp]
+    }
+  }
 
   return []
 }
 
 // Find possible values for property clauses in a and clause.
-export const getPropertyClausesStrings = node => {
+export const getPropertyClausesStringsOrRegex = node => {
```

Comment:
IMHO, we can keep the old name as we only handle *string* regexps, not all regexps.